garrysmod/gamemodes/terrortown/entities/entities/ttt_health_station.lua

```diff
@@ -171,5 +146,31 @@ if SERVER then
          nextcharge = CurTime() + self.RechargeFreq
       end
    end
-end
 
+   local ttt_detective_dmg_selfhs = CreateConVar("ttt_detective_dmg_selfhs", "0") -- 0 as detective cannot damage their own health station
+	
+   -- traditional equipment destruction effects
+   function ENT:OnTakeDamage(dmginfo)
+      if dmginfo:GetAttacker() == self:GetPlacer() and not ttt_detective_dmg_selfhs:GetBool() then return end
+   
+      self:TakePhysicsDamage(dmginfo)
+
+      self:SetHealth(self:Health() - dmginfo:GetDamage())
+
+      local att = dmginfo:GetAttacker()
+      local placer = self:GetPlacer()
+      if IsPlayer(att) then
+         DamageLog(Format("DMG: \t %s [%s] damaged health station [%s] for %d dmg", att:Nick(), att:GetRoleString(), placer:Nick(), dmginfo:GetDamage()))

```

Comment:
You'll have to check here if `IsPlayer(placer)`, because that player may have disconnected after placing the health station. If I recall correctly, other damagelog messages tend to print something like "disconnected" as the nickname if that happens.